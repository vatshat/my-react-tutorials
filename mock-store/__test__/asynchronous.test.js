import {
    fetchTodos,
    addTodos,
    setFetch,
} from '../src/asynchronous';

// set up redux-mock-store
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

it('should dispatch correct actions when fetching todos', () => {
    const store = mockStore({});

    const fakeFetch = () => {
        return Promise.resolve({
            json: () => Promise.resolve({
                todos: [{
                    name: 'fluffykins'
                }]
            })
        })
    }
    
    return store.dispatch(fetchTodos(fakeFetch))
        .then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
});