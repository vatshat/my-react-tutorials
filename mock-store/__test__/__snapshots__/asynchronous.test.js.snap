// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should dispatch correct actions when fetching todos 1`] = `
Array [
  Object {
    "isFetching": true,
    "type": "SET_FETCH",
  },
  Object {
    "todos": Object {
      "json": [Function],
    },
    "type": "ADD_TODOS",
  },
  Object {
    "isFetching": false,
    "type": "SET_FETCH",
  },
]
`;
